### ISSUE TITLE
zmq_z85_decode() doesn't catch invalid z85 chars within base85 range

### ISSUE THREAD

**Issue by TLINDEN** [zeromq/libzmq#903]
Not sure if this is a bug, but at least it's misleading.

Given the following test program:

```
char tilde[6] = "~~~~~";
byte dst[4];
int it
if(zmq_z85_decode(dst, tilde) == NULL) {
  fprintf(stderr, "fail\n");
}
else {
  for(i=0; i<4;++i) fprintf(stderr, "%02x", dst[i]);
  fprintf(stderr, "\n");
}
```

This prints `00000000`. The same works with `'''''` for example. It's misleading since characters such as `~` are not part of the z85 character base and should therefore not produce a 0, but an decoder error.

Or so I think.

best,
Tom

**Comment by sigiesec** [comment]
I think this was fixed by #2322

**Comment by bluca** [comment]
Yes, I think so. Closing, if it's an issue again we can reopen.
